flipperdevices/flipperzero-firmware
# [FL-3897] Happy mode

# What's new
Added a mode in which the dolphin never gets angry, as requested in #3843. This mode can be toggled via Settings > Desktop > Happy Mode with different screens displayed depending on whether the setting is currently enabled or disabled:
![Happy mode enabled](https://github.com/user-attachments/assets/621b348b-f36a-4f2d-9f1a-8ce3b176dac7)
![Happy mode disabled](https://github.com/user-attachments/assets/6113fefc-8742-4bc4-bbf0-2bb4522bf614)

This is implemented by always reporting the internal `butthurt` counter as being at zero when the setting is enabled.

In addition, the "Power OFF" screen will never show the sad dolphin when this mode is enabled:
![Power off screen in Happy mode](https://github.com/user-attachments/assets/35b0b091-7b7a-40d8-abd5-7508a8282872)
![Power off screen with Happy mode switched off](https://github.com/user-attachments/assets/e7aa98f4-9c35-4175-86e6-ee6a0bbf650e)


# Verification 
  - Build and flash the firmware with `DOLPHIN_DEBUG` in order to decrease waiting times
  - Wait for the dolphin to get angry
  - Turn on the setting and return to the desktop
  - Verify that the animation is now one of the happy ones
  - Verify that it now says "Mood: Happy" in the passport
  - Verify that the Power OFF screen does not show the sad dolphin image and text
  - Turn the setting back off again
  - Verify that the desktop animation is now one of the sad/angry ones
  - Verify that it now says "Mood: Angry" in the passport
  - Verify that the Power OFF screen now does show the sad dolphin image and text

# Checklist (For Reviewer)
